dll/ntdll/ldr/ldrutils.c

```diff
@@ -2454,6 +2455,14 @@ LdrpLoadDll(IN BOOLEAN Redirected,
     RtlInitEmptyUnicodeString(&RawDllName, NameBuffer, sizeof(NameBuffer));
     RtlCopyUnicodeString(&RawDllName, DllName);
 
+    /* Convert forward slashes to backslashes */
+    endptr = &RawDllName.Buffer[RawDllName.Length / sizeof(WCHAR) - 1];
+    for (pch = RawDllName.Buffer; pch != endptr; ++pch)
```

Comment:
On Windows, use a sane debugger (WinDbg) and put a breakpoint in ntdll!LdrpLoadDll. Then tell me whether the path string that is given to the function, already has its forward-slashes converted to backwards ones, or not.

i'm asking this, because you put this "fix" here, but LdrpLoadDll is a helper that is called by other functions, and it is as much probable it's one (or more) of these other functions that perform the change before calling the LdrpLoadDll.

Reply:
The testcase says Windows `LdrLoadDll` can accept forward slashes.

Reply:
It says that the _**upper-level exported function**_  LdrLoadDll can accept forward slashes, perhaps. But it doesn't say at which level the forward slashes are transformed.

Reply:
What is the lower level than `LdrLoadDll`?

Reply:
As you can see:
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/ntdll/ldr/ldrapi.c;hb=c07eb9fb1b7f3450e0806980cc0decf35cc73a50#l312
The "upper-level" LdrLoadDll() calls a lot of functions that act on the DllName, before calling at some point the "lower-level" LdrpLoadDll() helper function.
In addition, LdrpLoadDll() is used in many more places in ntdll:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=LdrpLoadDll
so modifying that one is potentially prone to introduce problems.

Reply:
https://jira.reactos.org/secure/attachment/67560/WinDbg-LdrLoadDll-got-forward-slashes.png
https://jira.reactos.org/secure/attachment/67561/WinDbg-LdrpLoadDll-got-forward-slashes.png